rust/rust.bzl

```diff
@@ -629,15 +663,14 @@ _rust_toolchain_attrs = {
 
 _rust_library_attrs = _rust_common_attrs + {
     "crate_type": attr.string(),
+    "platform": attr.string(),
 }
 
 rust_library = rule(
     _rust_library_impl,
     attrs = _rust_library_attrs + _rust_toolchain_attrs,
     fragments = ["cpp"],
-    outputs = {
-        "rust_lib": "lib%{name}.rlib",
-    },
+    outputs = _get_crate_type_and_target_outputs,
```

Comment:
I didn't implement the feature (it has been in Bazel since April 2015), but I have a pending change that improves the documentation for implicit outputs in general (including callbacks).

Currently, the documentation of the "outputs" parameter mentions callbacks:
https://bazel.build/versions/master/docs/skylark/lib/globals.html#rule